src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,86 @@
+package com.internet.shop.web.filters;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter("/*")
+public class AuthorizationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/admin/actions", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/users/orders", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/products/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/orders/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/orders/delete", List.of(Role.RoleName.ADMIN));
+
+        protectedUrls.put("/orders/complete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/get", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/users/orders", List.of(Role.RoleName.USER));
+        protectedUrls.put("/products/all", List.of(Role.RoleName.USER));
```

Comment:
I think this last one can be accessed by anyone, even if the user is not authenticated

Reply:
There are two controllers for showing all products and two jsp files for each of them:
1) "/products/all" -> products/all.jsp   - for users, with button "Buy" near each product
2) "/admin/products/all" -> admin/products/all.jsp   - for admins, with button "Delete" near each product and button "Add new product" below the table.

It was task of 26.2 homework https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/homework/lesson_16_hw2.html

So I think this two controllers should have different access, one for users and other for admins.

Reply:
okay :)